app/javascript/components/mentoring/mentor_inbox.jsx

```diff
@@ -1,75 +1,33 @@
-import React, { useReducer, useCallback } from 'react'
+import React from 'react'
 import { ConversationList } from './mentor_inbox/conversation_list'
-import { ConversationFilter } from './mentor_inbox/conversation_filter'
-import { ConversationSorter } from './mentor_inbox/conversation_sorter'
+import { TextFilter } from './text_filter'
+import { Sorter } from './sorter'
 import { TrackFilter } from './mentor_inbox/track_filter'
-
-function reducer(state, action) {
-  switch (action.type) {
-    case 'page.changed':
-      return { ...state, query: { ...state.query, page: action.payload.page } }
-    case 'track.changed':
-      return {
-        ...state,
-        query: { ...state.query, track: action.payload.track, page: 1 },
-      }
-    case 'sort.changed':
-      return {
-        ...state,
-        query: { ...state.query, sort: action.payload.sort },
-      }
-    case 'filter.changed':
-      return {
-        ...state,
-        query: { ...state.query, filter: action.payload.filter, page: 1 },
-      }
-    default:
-      if (process.env.NODE_ENV === 'development') {
-        throw new Error(`Unknown action type: ${action.type}`)
-      }
-      return state
-  }
-}
+import { useList } from '../../hooks/use_list'
 
 export function MentorInbox({ tracksRequest, sortOptions, ...props }) {
-  const [conversationsRequest, dispatch] = useReducer(
-    reducer,
-    Object.assign({ query: { page: 1 } }, props.conversationsRequest)
+  const [conversationsRequest, setFilter, setSort, setPage, addQuery] = useList(
+    props.conversationsRequest
   )
 
-  const setPage = useCallback(
-    (page) => {
-      dispatch({ type: 'page.changed', payload: { page: page } })
-    },
-    [dispatch]
-  )
-
-  const setTrack = useCallback(
-    (track) => {
-      dispatch({ type: 'track.changed', payload: { track: track } })
-    },
-    [dispatch]
-  )
-
-  function setSort(sort) {
-    dispatch({ type: 'sort.changed', payload: { sort: sort } })
-  }
-
-  function setFilter(filter) {
-    dispatch({ type: 'filter.changed', payload: { filter: filter } })
+  const setTrack = (track) => {
+    addQuery({ track: track, page: 1 })
   }
 
   return (
-    <div>
+    <div className="mentor-inbox">
       <TrackFilter request={tracksRequest} setTrack={setTrack} />
-      <ConversationFilter
+      <TextFilter
         filter={conversationsRequest.query.filter}
         setFilter={setFilter}
+        id="conversation-filter"
```

Comment:
Slightly off-topic: do we have a naming strategy for IDs and classes?

Reply:
Please can we use classes everywhere rather than ids if possible (I think I mentioned this in another PR too). We should always presume we might get multiple of these components on a page, and ids should be unique.

In terms of naming strategy, we use hyphens not underscores. I'm very big on scoping with CSS and JS. So that a component's class should have a prefix (eg `co-...`) and then everything that every acts on it should be scoped to be `.co-mentoring-inbox .conversation-filter`. This guarantees that there is no leakage of styles etc between components.

Reply:
- For pages we'd then use `#page-...` for example. 
- For things that are global I tend to use the `.lo-` prefix (for layout)

Happy to change the prefixes, but would like them there.

Reply:
Cool. That's very convenient. This should probably be added to https://github.com/exercism/v3-website/pull/55/files

Reply:
Merged that so that we can PR directly to that file and have discussions (like this one) there.

Reply:
This has a label and thus needs an ID. Thanks for the reminder above, though.